bot/exts/help_channels/_cog.py

```diff
@@ -535,3 +540,94 @@ async def update_available_help_channels(self) -> None:
         )
         self.dynamic_message = new_dynamic_message["id"]
         await _caches.dynamic_message.set("message_id", self.dynamic_message)
+
+    @staticmethod
+    def _serialise_session_participants(participants: set[int]) -> str:
+        """Convert a set to a comma separated string."""
+        return ','.join(str(p) for p in participants)
+
+    @staticmethod
+    def _deserialise_session_participants(s: str) -> set[int]:
+        """Convert a comma separated string into a set."""
+        return set(int(user_id) for user_id in s.split(",") if user_id != "")
+
+    @lock.lock_arg(NAMESPACE, "message", attrgetter("channel.id"))
+    @lock.lock_arg(NAMESPACE, "message", attrgetter("author.id"))
+    async def notify_session_participants(self, message: discord.Message) -> None:
+        """
+        Check if the message author meets the requirements to be notified.
+
+        If they meet the requirements they are notified.
+        """
+        if await _caches.claimants.get(message.channel.id) == message.author.id:
+            return  # Ignore messages sent by claimants
+
+        if not await _caches.help_dm.get(message.author.id):
+            return  # Ignore message if user is opted out of help dms
+
+        if message.content == f"{self.bot.command_prefix}close":
+            return  # Ignore messages that are closing the channel
+
+        session_participants = self._deserialise_session_participants(
+            await _caches.session_participants.get(message.channel.id) or ""
+        )
+
+        if message.author.id not in session_participants:
+            session_participants.add(message.author.id)
+
+            user = self.bot.get_user(message.author.id)
+
+            embed = discord.Embed(
+                title="Currently Helping",
+                description=f"You're currently helping in <#{message.channel.id}>",
+                color=discord.Colour.green(),
+                timestamp=message.created_at
+            )
+            embed.add_field(name="Conversation", value=f"[Jump to message]({message.message.jump_url})")
+            await user.send(embed=embed)
+
+            await _caches.session_participants.set(
+                message.channel.id,
+                self._serialise_session_participants(session_participants)
+            )
+
+    @commands.group(name="helpdm")
+    async def help_dm_group(self, ctx: commands.Context) -> None:
+        """
+        User will receive a embed when they are "helping" in a help channel.
+
+        If they have helpdms off they will won't receive an embed.
+        If they have helpdms on they will receive an embed.
+        """
+        if ctx.invoked_subcommand is None:
+            await ctx.send_help(ctx.command)
+
+    @help_dm_group.command(name="on")
```

Comment:
There's quite a bit of repeated code in these two subcommands, I think they could both be brought into a single command which takes an argument of on/off.

See bot.converters.allowed_strings on how to limit input like that.

you could do something like 
```py
      requested_state_bool = requested_state.lower() == "on"
      if requested_state_bool == await _caches.help_dm.get(ctx.author.id, False):
```
where requested_state is the string yes or no from the function args. This if will tell you if they are trying to set their state to whats already stored, so you can show them the error, if not, then you can set their state to that

Reply:
So quick question, why does `await _caches.help_dm.get(ctx.author.id, False)` return true. I am testing it out and it works, but I don't understand why that would return true.

Reply:
The second parameter defines what to return if the key isn't found in redis. So if `ctx.author.id` is in there, it will return the value, if it's not, it will return False

Reply:
Before I push this, should I change anything? (I am going to add logging and the docstring)
```py
@commands.command(name="helpdm")
    async def helpdm_command(
            self,
            ctx: commands.Context,
            state: allowed_strings("on", "off") = None
    ) -> None:
        state_bool = state.lower() == "on"

        requested_state_bool = state.lower() == "on"
        if requested_state_bool == await _caches.help_dm.get(ctx.author.id, False):
            if await _caches.help_dm.get(ctx.author.id):
                await ctx.send(f"{constants.Emojis.cross_mark}{ctx.author.mention} Help DMs are already ON!")
                return

            if not await _caches.help_dm.get(ctx.author.id):
                await ctx.send(f"{constants.Emojis.cross_mark}{ctx.author.mention} Help DMs are already OFF!")
                return

        if state_bool:
            await _caches.help_dm.set(ctx.author.id, True)

            await ctx.send(f"{constants.Emojis.ok_hand} {ctx.author.mention} Help DMs ON!")
            return

        await _caches.help_dm.delete(ctx.author.id)

        await ctx.send(f"{constants.Emojis.ok_hand} {ctx.author.mention} Help DMs OFF!")
```

Reply:
That's pretty close to what I'd do, just 3 things
1. I don't think making the default value be None is needed, we want the user to be required to give this arg.
2. The extra if statements aren't needed inside the `if requested_state_bool == await _caches.help_dm.get(ctx.author.id, False):` line. As this if statement is only entered if they are trying to set their status to what's already stored. So this could be simplified to just 
```py
if requested_state_bool == await _caches.help_dm.get(ctx.author.id, False):
    await ctx.send(f"{constants.Emojis.cross_mark} {ctx.author.mention} Help DMs are already {state}!")
    return
```
3. The last block could also be simplified to something like
```py
if state_bool:
    await _caches.help_dm.set(ctx.author.id, True)
else:
    await _caches.help_dm.delete(ctx.author.id)
await ctx.send(f"{constants.Emojis.ok_hand} {ctx.author.mention} Help DMs {state}!")
```